### ISSUE TITLE
Samples Fail To Compile

### ISSUE THREAD

**Issue by musictopia2** [dotnet/maui-samples#177]
![problem](https://user-images.githubusercontent.com/39104606/123729512-f20bbd80-d849-11eb-9a71-9f555d6f7b9d.png)
After I download the samples and try to compile, here is exactly what happens.
I even tried to rerun the maui check and it said everything was fine.  I hope there is a way to fix it.  Would be disappointed if maui can't work on some computers.

**Comment by jonathanpeppers** [comment]
Does it work command-line with `dotnet build`?

If not, can you add `-bl` and zip + upload the `msbuild.binlog` file it creates here?

**Comment by musictopia2** [comment]
![problem2](https://user-images.githubusercontent.com/39104606/123731029-3c8e3980-d84c-11eb-9707-404c93158742.png)
I tried via command line and here is the errors that came up.  I can next try the extra argument and see what happens.

**Comment by musictopia2** [comment]
[build.zip](https://github.com/dotnet/maui-samples/files/6730345/build.zip)
Here is the build.zip now.

**Comment by jonathanpeppers** [comment]
The build log says you don't have the Android workload installed.

Can you try updating `maui-check` to latest (0.5.6) and running it again?
```cmd
> dotnet tool update -g redth.net.maui.check --version 0.5.6
> maui-check
```

**Comment by musictopia2** [comment]
I tried and same thing.  The first thing I tried to run was the maui check.  I then ran the tool to update then run again.  Still said everything was fine even though it was not.  Is there another way to install the android stuff for .net 6.  There was one point where everything was working.  Last week, I accidently installed everything from the C drive as templates.  Once that happened, I tried to do a windows restore.  After that, then it did not fix the problem.  Found I was able to uninstall templates.  So I uninstalled all the ones from the C drive.  After that, then the wpf blazor did not work.  So I did a restore to undo the previous one.  Thought everything was fine until now.  I even tried to repair .net 6 preview 5.  Finally, I uninstalled preview 5 and reinstalled it.

**Comment by jonathanpeppers** [comment]
Did you get it working or not?

Can you try: https://github.com/dotnet/maui/blob/main/src/Workload/README.md#cleanup-net-6-installs--workloads

You shouldn't need to do anything with Windows restore, just delete some folders and run `maui-check` again.

**Comment by musictopia2** [comment]
After doing that, it was finally successful.  The first time I did windows restore was to fix a problem with installing lots of templates i did not want.  when it did not undo it, i figured out how to manually uninstall the templates.  Then wpf got hosed.  undoing the windows restore fixed the wpf blazor problem.  However, this problem occurred.  Its too bad that doing a repair would not have fixed the problem.  Thanks for the link.  Because I never would have guessed how to not only uninstall manually but also manually delete folders that was left behind.

**Comment by jonathanpeppers** [comment]
There are some known issues with the installer when upgrading from Preview 4 to Preview 5:

https://github.com/dotnet/core/blob/main/release-notes/6.0/known-issues.md#preview-5

It sounds like you got it working, though. Thanks!

**Comment by musictopia2** [comment]
I never upgraded from 4 to 5.  I actually upgraded from 3 to 5.  I skipped preview 4.  At first, preview 5 worked.  The problems occured when I made a mistake with installing templates and tried windows restore to fix the problem.  The windows restore corrupted the .net 6 and still did not fix the problem I was experiencing.  I appreciate the link that showed how to do a fresh install of .net 6.  Too bad doing a repair could not have done the same thing.  Because at first, I did a repair, even uninstall and install and even a maui check.  Never would have guessed that some folders had to be deleted manually.   When I first experienced it, if I had that link, then that would have fixed it even last night.
